### ISSUE TITLE
Support for commands with a variable number of parameters, e.g. HMSET

### ISSUE THREAD

**Issue by Dowwie** [benashford/redis-async-rs#22]
Hi Ben

Is the HMSET operation supported?  This doesn't seem to work, using a Vec<(String, String)> type:

```rust

        let items: Vec<(String, String)> = vec![("one".to_string(), "uno".to_string()),
                                               ("two".to_string(), "dos".to_string()),
                                               ("three".to_string(), "tres".to_string())];
        
        resp_array!["HMSET", "mykey", items]))
```

**Update**:  Unfortunately, the solution is a bit more challenging:
```rust
    let items = vec![("one".to_string(), "uno".to_string()),
                     ("two".to_string(), "dos".to_string()),
                     ("three".to_string(), "tres".to_string())];

    let parts: Vec<RespValue> = 
        vec!["HMSET".into(), "mykey".into()].into_iter()
        .chain(items.into_iter().flat_map(|(a, b)| once(a.into()).chain(once(b.into()))))
        .collect();
    
    let send_fut = redis.send(Command(RespValue::Array(parts)));
```

**Comment by benashford** [comment]
Yes, this is a bit of an edge case, when the number of parameters in a Redis command is variable it's a little bit ergonomic (well, quite painful) to construct the command.

If you don't mind, I'm going to re-open this issue as a reminder that a better way of handling these situation is needed.

**Comment by benashford** [comment]
I've just published version `0.4.1` which includes an `append` function to aide this type of scenario.

So now it's possible to do:

```rust
resp_array!["RPUSH", "keyname"].append(vector_of_values)
```

in addition to using `resp_array!` the usual way.

**Comment by Dowwie** [comment]
@benashford that's much better!
